drivers/spi/spi_context.h

```diff
@@ -263,7 +269,13 @@ void spi_context_update_rx(struct spi_context *ctx, u8_t dfs, u32_t len)
 static ALWAYS_INLINE
 bool spi_context_rx_on(struct spi_context *ctx)
 {
-	return !!(ctx->rx_buf || ctx->rx_len);
+	return !!(ctx->rx_len);
+}
+
+static ALWAYS_INLINE
+bool spi_context_rx_buf_on(struct spi_context *ctx)
+{
+	return !!(ctx->rx_buf && ctx->rx_len);
```

Comment:
Hi @tbursztyka this might break the stm32 driver.  Inside the shift-next-byte loop, the stm32 does the following:

```
data = read byte
if spi_context_rx_on(...) {
   *ctx.rx_buf = data
   spi_context_update_rx(...)
}
```
With this change, the above code will fail when `rx_buf.buf = NULL` and `rx_buf.len > 0`.

WDYT about documenting the spi_context API then we can go through and fix the current users?

Reply:
I have already push a commit to fix it in PR #5413.

Reply:
Ack